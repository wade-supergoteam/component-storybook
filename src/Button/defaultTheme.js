// @flow

export const primary = "#007bff";
export const secondary = "#6c757d";
export const success = "#28a745";
export const info = "#17a2b8";
export const warning = "#ffc107";
export const danger = "#dc3545";
export const light = "#f8f9fa";
export const dark = "#343a40";

export const fontSizeBase = "1rem";
export const fontSizeLg = "1.25rem";
export const fontSizeSm = ".875rem";

export const enableShadows = false;

export const btnPaddingY = ".375rem";
export const btnPaddingX = ".75rem";
export const btnLineHeight = "1.5";
export const btnPaddingYsm = ".25rem";
export const btnPaddingXsm = ".5rem";
export const btnLineHeightSm = "1.5";
export const btnPaddingYlg = ".5rem";
export const btnPaddingXlg = "1rem";
export const btnLineHeightLg = "1.5";
export const btnBorderWidth = "1px";
export const btnFontWeight = "400";
export const btnBoxShadow = "inset 0 1px 0 rgba(255,255,255,0.15),0 1px 1px rgba(0,0,0,0.075)";
export const btnFocusWidth = ".2rem";
export const btnFocusBoxShadow = "0 0 0 .2rem rgba(0,123,255,0.25)";
export const btnDisabledOpacity = ".65";
export const btnActiveBoxShadow = "inset 0 3px 5px rgba(0,0,0,0.125)";
export const btnLinkDisabledColor = "#6c757d";
export const btnBlockSpacingY = ".5rem";
export const btnBorderRadius = ".25rem";
export const btnBorderRadiusLg = ".3rem";
export const btnBorderRadiusSm = ".2rem";
export const btnTransition =
  "color .15s ease-iOut,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out";
