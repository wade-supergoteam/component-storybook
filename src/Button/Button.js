// @flow

import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { boxShadow } from "../utils/box-shadow";
import { hover, hoverFocus } from "../utils/hover";
import themeProp from "../utils/theme";
import { transition } from "../utils/transition";
import * as defaultTheme from "./defaultTheme";
import { buttonOutlineVariant } from "./utils/button-outline-variant";
import { buttonSize } from "./utils/button-size";
import { buttonVariant } from "./utils/button-variant";

const {
  enableShadows,
  btnBoxShadow,
  btnBorderWidth,
  btnFocusBoxShadow,
  btnDisabledOpacity,
  btnPaddingY,
  btnPaddingX,
  fontSizeBase,
  btnLineHeight,
  btnBorderRadius,
  btnPaddingYlg,
  btnPaddingXlg,
  fontSizeLg,
  btnLineHeightLg,
  btnBorderRadiusLg,
  btnPaddingYsm,
  btnPaddingXsm,
  fontSizeSm,
  btnLineHeightSm,
  btnBorderRadiusSm,
  btnTransition,
} = defaultTheme;

const Button = styled.button`
  display: ${({ block }) => (block ? "block" : "inline-block")};
  width: ${({ block }) => block && "100%"};
  font-weight: ${themeProp("btnFontWeight")};
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: ${themeProp("btnBorderWidth", btnBorderWidth)} solid transparent;

  ${transition(themeProp("btnTransition", btnTransition))};

  /* Share hover and focus styles */
  ${hoverFocus("text-decoration: none;")};

  &:focus,
  &.focus {
    outline: 0;
    box-shadow: ${themeProp("btnFocusBoxShadow", btnFocusBoxShadow)};
  }

  &.disabled,
  &:disabled {
    opacity: ${themeProp("btnDisabledOpacity", btnDisabledOpacity)};
    ${boxShadow("none")};
  }

  /* Opinionated: add "hand" cursor to non-disabled .btn elements */
  &:not(:disabled):not(.disabled) {
    cursor: pointer;
  }

  ${({ theme, color }) =>
    color !== "link" &&
    buttonVariant(
      theme[color] || defaultTheme[color],
      theme[color] || defaultTheme[color],
      theme.enableShadows || enableShadows,
      theme.btnBoxShadow || btnBoxShadow,
      theme.btnActiveBoxShadow,
    )};

  ${({ color }) =>
    color === "link" &&
    css`
      font-weight: ${themeProp("fontWeightNormal", "normal")};
      color: ${themeProp("linkColor", "#007bff")};
      background-color: transparent;



      ${hover(css`
        color: ${themeProp("linkHoverColor", "#0056b3")};
        text-decoration: ${themeProp("linkHoverDecoration", "underline")};
        background-color: transparent;
        border-color: transparent;
      `)};

      &:focus,
      &.focus {
        text-decoration: ${themeProp("linkHoverDecoration", "underline")};
        border-color: transparent;
        box-shadow: none;
      }

      &:disabled,
      &.disabled {
        color: text-decoration: ${themeProp("btnLinkDisabledColor", "#6c757d")};
      }
    `};

  ${({ theme, outline, color }) => outline && buttonOutlineVariant(theme[color] || defaultTheme[color])};

  ${({ size, theme }) =>
    size === "normal" &&
    buttonSize(
      themeProp("btnPaddingY", btnPaddingY),
      themeProp("btnPaddingX", btnPaddingX),
      themeProp("fontSizeBase", fontSizeBase),
      themeProp("btnLineHeight", btnLineHeight),
      typeof theme.enableRounded === "undefined" || theme.enableRounded ? themeProp("btnBorderRadius", btnBorderRadius) : 0,
    )};

  ${({ size, theme }) =>
    size === "small" &&
    buttonSize(
      themeProp("btnPaddingYsm", btnPaddingYsm),
      themeProp("btnPaddingXsm", btnPaddingXsm),
      themeProp("fontSizeSm", fontSizeSm),
      themeProp("btnLineHeightSm", btnLineHeightSm),
      typeof theme.enableRounded === "undefined" || theme.enableRounded ? themeProp("btnBorderRadiusSm", btnBorderRadiusSm) : 0,
    )};

  ${({ size, theme }) =>
    size === "large" &&
    buttonSize(
      themeProp("btnPaddingYlg", btnPaddingYlg),
      themeProp("btnPaddingXlg", btnPaddingXlg),
      themeProp("fontSizeLg", fontSizeLg),
      themeProp("btnLineHeightLg", btnLineHeightLg),
      typeof theme.enableRounded === "undefined" || theme.enableRounded ? themeProp("btnBorderRadiusLg", btnBorderRadiusLg) : 0,
    )};

  ${({ active }) =>
    active &&
    css`
      background-image: none;
      ${boxShadow(themeProp("btnFocusBoxShadow", btnFocusBoxShadow))};
    `};
`;

// Button.Link = Button.withComponent('a');

Button.defaultProps = {
  active: false,
  block: false,
  color: "primary",
  size: "normal",
  outline: false,
};

Button.propTypes = {
  active: PropTypes.bool,
  block: PropTypes.bool,
  color: PropTypes.string,
  size: PropTypes.string,
  outline: PropTypes.bool,
};

export default Button;
