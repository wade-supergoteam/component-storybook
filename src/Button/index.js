import Button from "./Button";
import { buttonVariant } from "./utils/button-variant";
import { buttonSize } from "./utils/button-size";
import { buttonOutlineVariant } from "./utils/button-outline-variant";

export { buttonVariant, buttonSize, buttonOutlineVariant };
export default Button;
