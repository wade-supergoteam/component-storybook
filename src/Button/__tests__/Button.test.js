import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

test("it works", () => {
  const wrapper = renderer.create(<Button />);
  expect(wrapper).toMatchSnapshot();
});
