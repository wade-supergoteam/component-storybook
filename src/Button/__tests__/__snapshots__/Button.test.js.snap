// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
<button
  className="sc-bdVaJa zXMdD"
  color="primary"
  size="normal"
/>
`;
